src/vs/workbench/browser/parts/editor/tabsTitleControl.ts

```diff
@@ -517,15 +529,17 @@ export class TabsTitleControl extends TitleControl {
 			oldOptions.pinnedTabSizing !== newOptions.pinnedTabSizing ||
 			oldOptions.showIcons !== newOptions.showIcons ||
 			oldOptions.hasIcons !== newOptions.hasIcons ||
-			oldOptions.highlightModifiedTabs !== newOptions.highlightModifiedTabs
+			oldOptions.highlightModifiedTabs !== newOptions.highlightModifiedTabs ||
+			oldOptions.multiLineTabs !== newOptions.multiLineTabs
 		) {
 			this.redraw();
 		}
+	}
 
-		// Udate tabs scrollbar sizing
-		if (oldOptions.titleScrollbarSizing !== newOptions.titleScrollbarSizing) {
-			this.updateTabsScrollbarSizing();
-		}
+	private updateMultiLineTabs(): void {
```

Comment:
There now seems to be a weird condition under which we seem to be adding the `multi-line` class when settings change but we conditionally add and remove it in the `layout` call. Shouldn't we exclusively be setting this class only from the layout call given that this is the only place where we know that we need it when the size is little?

I would say we let the `layout` method take care and make sure it is called when this setting changes.

Reply:
Ah yes, I forgot about that when moving it to the `layout` method. I've removed it now, thanks.